### PR TITLE
Run integration tests on every build

### LINKED ISSUE #80: Run integration tests during every build
Either using an always available cluster or spin one up for the tests.

Always on cluster (GKE):
- Constant payments for infrastructure
+ Faster tests
+ Tests upgrades
+ Tests long-running operator

Ad-hoc cluster:
+ Free
- Slow to spin up
- Only tests freshly spun up operator
- Memory leaks and other issues of long-running process don't get tested

### PR DESCRIPTION
**Pull request by psycho-ir**
closes #80

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,17 +11,27 @@ jobs:
...
           restore-keys: |
-            ${{ runner.os }}-maven-  
+            ${{ runner.os }}-maven-
       - name: Set up JDK
         uses: actions/setup-java@v1
         with:
           java-version: ${{ matrix.java }}
-      - name: Build with Maven
-        run: mvn -B package -P no-integration-tests --file pom.xml
+      - name: Run unit tests
```
Comment (psycho-ir) [inline .github/workflows/ci.yml:27]: Running unit tests before setting up the minikube help to get feedback from unit tests quicker and continue only if all unit tests passed

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -11,17 +11,27 @@ jobs:
...
+        kubernetes: ['v1.17.13','v1.18.12','v1.19.4']
     steps:
       - uses: actions/checkout@v2
       - uses: actions/cache@v2
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
-            ${{ runner.os }}-maven-  
+            ${{ runner.os }}-maven-
```
Comment (psycho-ir) [inline .github/workflows/ci.yml:22]: Do you think we need those 2 spaces?

Comment (csviri) [reply .github/workflows/ci.yml:22]: probably not

**Comment by csviri** [comment]
LGTM
thank you @psycho-ir !

**Pull request merged**